test/test_integration_cluster.rb

```diff
@@ -387,14 +387,36 @@ def test_fork_worker_phased_restart_with_high_worker_count
       workers #{worker_count}
     CONFIG
 
-    # workers is the default
     get_worker_pids 0, worker_count
 
     Process.kill :USR1, @pid
 
     get_worker_pids 1, worker_count
 
-    # below is so all of @server_log isn't output for failure
+    # below is so all of @server_log isn't simply output on failure
+    refute @server_log[/.*Terminating timed out worker.*/]
+  end
+
+  def test_refork_phased_restart_with_fork_worker_and_high_worker_count
+    worker_count = 10
+
+    cli_server "test/rackup/hello.ru", config: <<~CONFIG
+      fork_worker 1000 # not going to be reached
+      worker_check_interval 1
+      # lower worker timeout from default (60) to avoid test timeout
+      worker_timeout 2
+      # to simulate worker 0 timeout, total boot time for all workers
+      # needs to exceed single worker timeout
+      workers #{worker_count}
+    CONFIG
+
+    get_worker_pids 0, worker_count
+
+    Process.kill :SIGURG, @pid
+
+    get_worker_pids 1, worker_count - 1
```

Comment:
I don't understand why we do `worker_count - 1`

Reply:
Because for `fork_worker` with two workers, worker 0 remains as is and a new worker 1 is forked from it after which the old worker 1 is terminated. Since that logic piggy backs off of phased restart the logs indicate that worker 1 is now in phase 2. We don't see any logs for worker 0 as expected, hence the `-1`.

In the previous test however we just do a plain phased restart, which will restart all workers, worker 0 included.

Reply:
Fork worker example:
<img width="1728" alt="Screenshot 2025-01-10 at 11 26 57 pm" src="https://github.com/user-attachments/assets/405a620a-522f-4ff3-8903-e8a15a254331" />

Reply:
Phased restart example:
<img width="1728" alt="Screenshot 2025-01-10 at 11 26 57 pm" src="https://github.com/user-attachments/assets/065b29c8-8001-4c53-bc6e-62b24dd1c2b2" />

Reply:
We should probably update the logs to mention fork worker and not phased restart on refork...